### PR TITLE
Cloudfront private key setting file or key content #254

### PR DESCRIPTION
**Pull request by gleimermora**

### CODE REVIEW

File: classes/local/manager.php [hunk 1]
```diff
@@ -196,20 +195,31 @@ static public function cloudfront_pem_exists() {
...
+    /**
+     * @param string $cloudfrontprivatekey
+     * @return bool|resource
+     */
+    static public function parse_cloudfront_private_key($cloudfrontprivatekey) {
+        global $CFG;
+        $path = $CFG->dataroot . $cloudfrontprivatekey;
+        if (file_exists($path) && is_readable($path)) {
+            $cert = file_get_contents($path);
+            $parsedkey = openssl_pkey_get_private($cert);
```
Comment (brendanheywood) [inline classes/local/manager.php]: why not close the if here and have the same code path after this?

File: lang/en/tool_objectfs.php [hunk 2]
```diff
@@ -162,11 +162,11 @@
 $string['settings:presignedcloudfronturl:cloudfront_key_pair_id'] = 'Key_Pair ID from AWS';
 $string['settings:presignedcloudfronturl:cloudfront_key_pair_id_help'] = 'This is generated using AWS account \'root\' user (along with the private key .pem file).';
 $string['settings:presignedcloudfronturl:cloudfront_private_key_pem'] = 'PRIVATE Key .pem';
-$string['settings:presignedcloudfronturl:cloudfront_private_key_pem_help'] = 'Private key content (.pem file).';
+$string['settings:presignedcloudfronturl:cloudfront_private_key_pem_help'] = 'Private key content (.pem file) or relative path to cloudfront.pem file. ($CFG->dataroot . "/cloudfront.pem")';
```
Comment (brendanheywood) [inline lang/en/tool_objectfs.php]: This string seems inconsistent with the implementation?

File: classes/local/manager.php (hunk 1 above)
Comment (gleimermora) [reply classes/local/manager.php]: done

File: lang/en/tool_objectfs.php (hunk 2 above)
Comment (gleimermora) [reply lang/en/tool_objectfs.php]: updated

**Pull request merged**